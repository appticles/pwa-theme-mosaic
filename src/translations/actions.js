import config from 'react-global-configuration';

export const REQUEST_TRANSLATIONS = 'REQUEST_TRANSLATIONS';
export const RECEIVE_TRANSLATIONS = 'RECEIVE_TRANSLATIONS';

export const requestTranslations = () => ({
  type: REQUEST_TRANSLATIONS,
});

export const receiveTranslations = translations => ({
  type: RECEIVE_TRANSLATIONS,
  translations,
});

export const fetchTranslations = (dispatch) => {
  dispatch(requestTranslations());
  config.set(window.__INITIAL_CONFIG__, { freeze: false });

  return fetch(config.get('translate').path, {
    method: 'GET',
    header: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      mode: 'cors',
    },
  })
    .then(response => response.json())
    .then(json => dispatch(receiveTranslations(json)))
    .catch(() => {
      dispatch(receiveTranslations([]));
    });
};
