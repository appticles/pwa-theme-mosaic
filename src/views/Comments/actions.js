import config from 'react-global-configuration';

export const REQUEST_COMMENTS = 'REQUEST_COMMENTS';
export const RECEIVE_COMMENTS = 'RECEIVE_COMMENTS';

export const requestComments = () => ({
  type: REQUEST_COMMENTS,
});

export const receiveComments = comments => ({
  type: RECEIVE_COMMENTS,
  comments,
});

export const fetchComments = (params = {}) => (dispatch) => {
  dispatch(requestComments());

  let url;
  if (params) {
    url =
      config.get('export').comments +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receiveComments(json)))
    .catch(() => {
      dispatch(receiveComments([]));
    });
};

export const addComment = (params = {}) => {
  fetch(config.get('export').comments, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ params }),
  });
};
