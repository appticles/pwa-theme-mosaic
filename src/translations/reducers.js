import { combineReducers } from 'redux';
import Immutable from 'seamless-immutable';

import { REQUEST_TRANSLATIONS, RECEIVE_TRANSLATIONS } from './actions';

export const INITIAL_STATE = Immutable({
  items: [],
  isFetching: 0,
});

const items = (state = INITIAL_STATE.items, action) => {
  switch (action.type) {
  case REQUEST_TRANSLATIONS:
    return state;
  case RECEIVE_TRANSLATIONS:
    return action.translations;
  default:
    return state;
  }
};
const isFetching = (state = INITIAL_STATE.isFetching, action) => {
  switch (action.type) {
  case REQUEST_TRANSLATIONS:
    return state + 1;
  case RECEIVE_TRANSLATIONS:
    return state - 1;
  default:
    return state;
  }
};

export const getTranslations = state => state.items;
export const getTranslationsFetching = state => state.isFetching;
export default combineReducers({ items, isFetching });
