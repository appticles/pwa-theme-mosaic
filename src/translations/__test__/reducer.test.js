import translations, { INITIAL_STATE, getTranslations, getTranslationsFetching } from '../reducers';
import { requestTranslations, receiveTranslations } from '../actions';

describe('Translations reducer', () => {
  it('returns the same state on an unhandled action', () => {
    expect(translations(INITIAL_STATE, { type: '_NULL' })).toMatchSnapshot();
  });
  it('handles REQUEST_TRANSLATIONS action', () => {
    const translationsList = {
      isFetching: 0,
      items: {
        TEXTS: 'Some text',
        FORM: 'Some text',
      },
    };
    expect(translations(translationsList, requestTranslations())).toMatchSnapshot();
  });
  it('handles RECEIVE_TRANSLATIONS action', () => {
    const translationsState = {
      isFetching: 1,
      items: {
        TEXTS: 'Some text',
        FORM: 'Some text',
      },
    };
    const translationsList = {
      isFetching: 1,
      items: {
        TEXTS: 'Some text',
        FORM: 'Some text',
      },
    };
    expect(translations(translationsState, receiveTranslations(translationsList))).toMatchSnapshot();
  });

  // reducer functions
  it('getTranslationsFetching returns state.isFetching', () => {
    const state = {
      isFetching: 4,
    };
    expect(getTranslationsFetching(state)).toBe(4);
  });
  it('getTranslations returns state.items', () => {
    const state = {
      items: {
        TEXTS: 'Some text',
        FORM: 'Some text',
      },
    };
    expect(getTranslations(state)).toBe(state.items);
  });
});
