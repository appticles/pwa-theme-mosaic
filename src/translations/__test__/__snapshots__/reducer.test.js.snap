// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translations reducer handles RECEIVE_TRANSLATIONS action 1`] = `
Object {
  "isFetching": 0,
  "items": Object {
    "isFetching": 1,
    "items": Object {
      "FORM": "Some text",
      "TEXTS": "Some text",
    },
  },
}
`;

exports[`Translations reducer handles REQUEST_TRANSLATIONS action 1`] = `
Object {
  "isFetching": 1,
  "items": Object {
    "FORM": "Some text",
    "TEXTS": "Some text",
  },
}
`;

exports[`Translations reducer returns the same state on an unhandled action 1`] = `
Object {
  "isFetching": 0,
  "items": Object {},
}
`;
